modules/end-user-guide/partials/proc_creating-workspaces.adoc

```diff
@@ -2,4 +2,215 @@
+        app.kubernetes.io/component: vscode-runtime
+        app.kubernetes.io/part-of: vscode.eclipse.org
+        controller.devfile.io/container-contribution: true
+      container:
+        cpuRequest: 30m
+        command:
+          - /checode/entrypoint-volume.sh
+        env:
+          - name: CODE_HOST
+            value: 0.0.0.0
+        memoryRequest: 256Mi
+        sourceMapping: /projects
+        cpuLimit: 500m
+        volumeMounts:
+          - name: checode
+            path: /checode
+        memoryLimit: 1024Mi
+        image: 'quay.io/che-incubator/che-code:insiders'
+        endpoints:
+          - attributes:
+              cookiesAuthEnabled: true
+              discoverable: false
+              type: main
+              urlRewriteSupported: true
+            exposure: public
+            name: che-code
+            protocol: https
+            secure: false
+            targetPort: 3100
+          - attributes:
+              discoverable: false
+              urlRewriteSupported: true
+            exposure: public
+            name: code-redirect-1
+            protocol: http
+            targetPort: 13131
+          - attributes:
+              discoverable: false
+              urlRewriteSupported: true
+            exposure: public
+            name: code-redirect-2
+            protocol: http
+            targetPort: 13132
+          - attributes:
+              discoverable: false
+              urlRewriteSupported: true
+            exposure: public
+            name: code-redirect-3
+            protocol: http
+            targetPort: 13133
+      name: vscode-runtime-description
+    - name: checode
+      volume:
+        ephemeral: true
+  events:
+    preStart:
+      - init-container-command
+----
+<1> Name of the `DevWorkspaceTemplate` custom resource. This name will be used to reference the `DevWorkspaceTemplate` custom resource within the `DevWorkspace` custom resource.
+<2> User namespace, which is the target {orch-namespace} for the new workspace.
+====
+
+. To prepare the `DevWorkspace` custom resource, copy the contents of the target Git repository's devfile.
++
+.Copied devfile contents with `schemaVersion: 2.2.0`
+====
+[source,yaml,subs="+quotes,+attributes"]
+----
+components:
+  - name: tooling-container
+    container:
+      image: quay.io/devfile/universal-developer-image:ubi8-latest
+----
+====
++
+TIP: For more details, see the link:https://devfile.io/docs/2.1.0/what-is-a-devfile[devfile v2 documentation].
+
+. Create a `DevWorkspace` custom resource, pasting the devfile contents from the previous step under the `spec.template` field.
++
+.A `DevWorkspace` custom resource
+====
+[source,yaml,subs="+quotes,+attributes"]
+----
+kind: DevWorkspace
+apiVersion: workspace.devfile.io/v1alpha2
+metadata:
+  name: my-devworkspace#<1>
+  namespace: user1-dev#<2>
+spec:
+  routingClass: che
+  started: true#<3>
+  contributions:#<4>
+    - name: ide
+      kubernetes:
+        name: che-code#<5>
```

Comment:
Does it make sense to also explain that you can skip the "creating a DWT" part if your resource is available online? Contributions can also be specified by URI, e.g. 
```yaml
contributions:
  - name: ide
    uri: <URL for devfile/devworkspace content for editor>
```

Reply:
This is a good idea. And also I was thinking that it will be hard to maintain the DWT sample. It would be better if we could just reference the DWT of `che-incubator/che-code/insiders` from the plugin registry. But I am afraid that we do not publish it (we only publish the `devfile.yaml`). @svor is that correct?

Reply:
Currently plugin reg publishes all editors as a devfile.yaml files: https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/insiders/devfile.yaml

Reply:
The URI field can be either a DevWorkspaceTemplate or Devfile, DWO handles both. This is used in the built-in DWO samples, for example: https://github.com/devfile/devworkspace-operator/blob/main/samples/theia-latest-contributions.yaml#L22

Reply:
Any thoughts on removing the DWT sample in the docs and replacing it with:
```
export DWT_SPEC=$(curl https://<che_fqdn>/plugin-registry/main/v3/plugins/che-incubator/che-code/insiders/devfile.yaml | yq 'del(.schemaVersion, .metadata)')

cat <<EOF | kubectl apply -f -
{
  "apiVersion":"workspace.devfile.io/v1alpha2",
  "kind":"DevWorkspaceTemplate",
  "metadata":{
    "name":"che-code"
  },
  "spec":$DWT_SPEC
}
EOF

```

so that we don't need to maintain it directly in the docs? The above is working for upstream and downstream

Reply:
Can't you reference the che-code `devfile.yaml` in the devworksapce`contribution.uri` as mentioned by @amisevsk?

Reply:
The only downside of this approach is that DWO will download the devfile.yaml on every reconcile (though this hasn't been a huge issue in the past). 

The benefit of referencing the devfile by URI is that it'll automatically get updates when Che is updated. With a manually-created DWT, it'll use a static UDI image unless you manually update it.

Reply:
@l0rd That works too, but IMHO I'm not sure if that method can replace the DWT sample in the docs, since DWT are used when creating workspaces from the dashboard and is an important workspace-related CR to know about

My new commit adds this tip, but still documents the DWT sample too:
![image](https://user-images.githubusercontent.com/83611742/214635936-b2d1b0ee-f816-447f-ae65-7700faa9a2d2.png)

Reply:
In this article we are introducing the DevWorkspace CR and that's already enough from my point of view. The DWT CR and how to define an editor is another concept and can be overwhelming. It would probably be worth a separate article so I would be happy if we could hide this complexity using the URI. 

And I like your approach that uses the link to the che-code definition of the particular Che plugin registry instance. I think that's better than using the link to the upstream online plugin registry che-code definition.

Reply:
Sounds good, thank you for the feedback, I've updated it to use the URI method cc @max-cx